frame/ethereum/src/lib.rs

```diff
@@ -244,12 +244,7 @@ impl<T: Config> frame_support::unsigned::ValidateUnsigned for Module<T> {
 
 			let mut builder = ValidTransactionBuilder::default()
 				.and_provides((origin, transaction.nonce))
-				.priority(if min_gas_price == U256::zero() {
-						0
-					} else {
-						let target_gas = (transaction.gas_limit * transaction.gas_price) / min_gas_price;
-						T::GasWeightMapping::gas_to_weight(target_gas.unique_saturated_into())
-				});
+				.priority(transaction.gas_price.unique_saturated_into());
```

Comment:
It may be better to still keep the gas weight mapping here.

```suggestion
				.priority(T::GasWeightMapping::gas_to_weight(transaction.gas_price.unique_saturated_into()));
```

Reply:
Is the idea that we should prioritize by weight price rather than gas price? That seems reasonable to me.

But I don't think that putting the gas _price_ through the gas weight mapping makes sense (except in the case of a linear mapping).

I'm okay making this change because it's better than what's there now and most chains probably do use a linear mapping. But as a followup we should simplify the mapping trait to enforce a linear mapping.

Reply:
Hmm I see what you mean. Yeah you're right!